coffee_core/src/coffee.rs

```diff
@@ -442,8 +448,70 @@ impl PluginManager for CoffeeManager {
         Err(err)
     }
 
-    async fn nurse(&mut self) -> Result<(), CoffeeError> {
-        self.recovery_strategies.scan().await
+    async fn nurse(&mut self) -> Result<CoffeeNurse, CoffeeError> {
+        let status = self.recovery_strategies.scan(self).await?;
+        let mut nurse_actions: Vec<NurseStatus> = vec![];
+        for defect in status.defects.iter() {
+            log::debug!("defect: {:?}", defect);
+            match defect {
+                Defect::RepositoryLocallyAbsent(repos) => {
+                    let mut actions = self.patch_repository_locally_absent(repos.to_vec()).await?;
+                    nurse_actions.append(&mut actions);
+                }
+            }
+        }
+
+        // If there was no actions taken by nurse, we return a sane status.
+        if nurse_actions.is_empty() {
+            nurse_actions.push(NurseStatus::Sane);
+        }
```

Comment:
It's a tradeoff
easier to read but after a second thought, I guess it makes more sense to return an empty vector because we are now returning a vector of actions taken and an empty vector makes more sense in that context 
I will consider this in the next PR for `nurse`